support/Environments/caltech_hpc_skylake_homegrown_gcc.sh

```diff
@@ -0,0 +1,88 @@
+    echo "run 'spectre_setup_modules' every time you log in:"
+    echo ""
+    echo "module use /central/groups/sxs/knelli/modules"
+}
+
+spectre_load_modules() {
+    module use /central/groups/sxs/knelli/modules
+    module load git/2.37.2
+    module load compilers/binutils/2.41
+    module load compilers/gcc/11.4.0
+    module load tools/cmake/3.28.1
+    module load tools/ccache/4.8.2
+    module load compilers/llvm/16.0.6 # For clang-format
+    module load libraries/blaze/3.8
+    module load libraries/boost/1.82.0
+    module load libraries/brigand/master
+    module load libraries/catch/3.4.0
+    module load oneapi/mpi-2021.11
+    module load libraries/charm/7.0.0
+    module load libraries/gsl/2.7
+    module load libraries/hdf5/1.12.3
+    module load libraries/jemalloc/5.3.0
+    module load libraries/libsharp/1.0.0
+    module load libraries/libxsmm/1.16.1-avx2
+    module load libraries/openblas/0.3.25-avx2
+    module load libraries/spec-exporter/2024-01
+    module load libraries/yaml-cpp/0.8.0
+    module load languages/python/spectre-python
+}
+
+spectre_unload_modules() {
+    module use /central/groups/sxs/knelli/modules
+    module unload git/2.37.2
+    module unload compilers/binutils/2.41
+    module unload compilers/gcc/11.4.0
+    module unload tools/cmake/3.28.1
+    module unload tools/ccache/4.8.2
+    module unload compilers/llvm/16.0.6 # For clang-format
+    module unload libraries/blaze/3.8
+    module unload libraries/boost/1.82.0
+    module unload libraries/brigand/master
+    module unload libraries/catch/3.4.0
+    module unload oneapi/mpi-2021.11
+    module unload libraries/charm/7.0.0
+    module unload libraries/gsl/2.7
+    module unload libraries/hdf5/1.12.3
+    module unload libraries/jemalloc/5.3.0
+    module unload libraries/libsharp/1.0.0
+    module unload libraries/libxsmm/1.16.1-avx2
+    module unload libraries/openblas/0.3.25-avx2
+    module unload libraries/spec-exporter/2024-01
+    module unload libraries/yaml-cpp/0.8.0
+    module unload languages/python/spectre-python
+}
+
+spectre_run_cmake() {
+    if [ -z ${SPECTRE_HOME} ]; then
+        echo "You must set SPECTRE_HOME to the cloned SpECTRE directory"
+        return 1
+    fi
+    spectre_load_modules
+
+    # Note that choosing the memory allocator to be JEMALLOC is important. When
+    # using the SYSTEM allocator, during BBH runs dumping volume data too often
+    # would result in memory corruption errors. Switching to JEMALLOC fixed
+    # these issues. It is still unclear *why* the SYSTEM allocator caused these
+    # issues. --Kyle Nelli
+    cmake -D CMAKE_C_COMPILER=gcc \
+          -D CMAKE_CXX_COMPILER=g++ \
+          -D CMAKE_Fortran_COMPILER=gfortran \
+          -D CHARM_ROOT=$CHARM_ROOT \
+          -D CMAKE_BUILD_TYPE=Release \
+          -D MEMORY_ALLOCATOR=JEMALLOC \
+          -D BUILD_PYTHON_BINDINGS=ON \
+          -D MACHINE=CaltechHpcSkylake \
```

Comment:
For now, yes I intend to have all three environments. However, the sys admins will be switching the entire machine over to a new OS soon (not exactly sure when) so the `caltech_hpc_skylake_gcc.sh` env will be obsolete because those are system installed modules on the current (soon to be old) OS. I plan to remove that env when the OS switch happens, then we'll just have skylake vs icelake with my installed modules.

As for the fftw module, that was needed because of the spec-exporter on the icelake nodes. The spec-exporter for skylake currently uses the system fftw (annoying). I can unify things when the OS switch happens.

For libbacktrace and paraview, I just didn't install those yet because they weren't of highest priority to get the code running. I can do those in a followup.

Reply:
quick note: the new [Paraview 5.12](https://www.paraview.org/download/) prebuilt binaries use py3.10, so should be compatible with your installation.

Reply:
I don't think we should use the prebuilt binaries. On wheeler for example the 5.11.1 paraview doesn't work on compute nodes, while the 5.10 version I compiled from scratch does. We really should have ParaView be built against the MPI and HDF5 we use, ideally the python too.